app/src/main/java/com/edu/uni/augsburg/uniatron/ui/onboarding/OnboardingActivity.java

```diff
@@ -0,0 +1,163 @@
+                                sharedPrefsHandler.setOnboardingAppUsageEntered();
+
+                                final DataRepository dataRepository = MainApplication.getRepository(
+                                        getApplicationContext());
+                                dataRepository.addAppUsage(getPackageName(), 60);
+                            }
+
+                            PermissionUtil.requestUsageAccess(getApplicationContext());
+                        }
+                    });
+        }
+        addSlide(userAccessSlideBuilder.build());
+    }
+
+    // COIN BAG & BATTERY OPTIMIZATION
+    private void addSlideCoinBag() {
+        final SimpleSlide.Builder coinbagSlideBuilder = new SimpleSlide.Builder();
+        coinbagSlideBuilder
+                .title(R.string.onboarding_coinbag)
+                .description(R.string.onboarding_coinbag_granted_description)
+                .image(R.drawable.ic_onboarding_coin)
+                .background(R.color.onboardingBackground6)
+                .backgroundDark(R.color.onboardingBackground6Dark)
+                .scrollable(true);
+
+        if (PermissionUtil.needBatteryWhitelistPermission(this)) {
+            coinbagSlideBuilder
+                    .description(R.string.onboarding_coinbag_description)
+                    .buttonCtaLabel(R.string.onboarding_btn_whitelist)
+                    .buttonCtaClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(final View view) {
+                            PermissionUtil.requestIgnoreBatterOptimization(getApplicationContext());
+                        }
+                    });
+        }
+        addSlide(coinbagSlideBuilder.build());
+    }
+
+    private void addSlideShop() {
+
+        addSlide(new SimpleSlide.Builder()
+                .title(R.string.onboarding_shop_title)
+                .description(R.string.onboarding_shop_description)
+                .image(R.drawable.ic_onboarding_shopping_cart)
+                .background(R.color.onboardingBackground4)
+                .backgroundDark(R.color.onboardingBackground4Dark)
+                .buttonCtaLabel(R.string.onboarding_btn_shop)
+                .buttonCtaClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(final View view) {
+
+                        // the user gets a step bonus on first onboarding so it's not empty
+                        final SharedPreferencesHandler sharedPrefsHandler =
+                                MainApplication.getSharedPreferencesHandler(getApplicationContext());
+
+                        if (sharedPrefsHandler.onboardingStepBonusEligible()) {
+                            sharedPrefsHandler.setOnboardingStepBonusGranted();
+
+                            final DataRepository dataRepository = MainApplication.getRepository(
```

Comment:
The database should only be called from a ViewModel. (see [example](https://github.com/FHellmann/Uniatron/blob/master/app/src/main/java/com/edu/uni/augsburg/uniatron/ui/MainActivityViewModel.java))